pkg/k8s/pod_utils.go

```diff
@@ -20,6 +24,24 @@ func GetPodCondition(pod *corev1.Pod, conditionType corev1.PodConditionType) *co
 	return nil
 }
 
+// PodNamespaceIndexer returns indexer to index Pods in datastore using namespace
+func PodNamespaceIndexer() cache.Indexers {
```

Comment:
let's have a NamespaceIndexFunc which uses meta.Accessor(obj), and it's not pod specific